cmd/workspace/apps/run-local.go

```diff
@@ -0,0 +1,205 @@
+
+	cmd.PreRunE = root.MustWorkspaceClient
+	cmd.RunE = func(cmd *cobra.Command, args []string) (err error) {
+		ctx := cmd.Context()
+		w := cmdctx.WorkspaceClient(ctx)
+		workspaceId, err := w.CurrentWorkspaceID(ctx)
+		if err != nil {
+			return err
+		}
+
+		cwd, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+
+		config := apps.NewConfig(w.Config.Host, workspaceId, cwd)
+		if entryPoint != "" {
+			config.AppSpecFiles = []string{entryPoint}
+		}
+		spec, err := apps.ReadAppSpecFile(config)
+		if err != nil {
+			return err
+		}
+
+		cfg := cmdctx.ConfigUsed(cmd.Context())
+		env := auth.ProcessEnv(cfg)
+		if cfg.Profile != "" {
+			env = append(env, "DATABRICKS_CONFIG_PROFILE="+cfg.Profile)
+		}
+
+		for _, envVar := range spec.EnvVars {
+			if envVar.Value != nil {
+				env = append(env, envVar.Name+"="+*envVar.Value)
+			}
+
+			if envVar.ValueFrom != nil {
+				if os.Getenv(envVar.Name) != "" {
+					customEnv = append(customEnv, envVar.Name+"="+os.Getenv(envVar.Name))
+				}
+				found := false
+				for _, e := range customEnv {
+					if strings.HasPrefix(e, envVar.Name+"=") {
+						found = true
+						break
+					}
+				}
+				if !found {
+					return fmt.Errorf("%s defined in app.yml with valueFrom property and can't be resolved locally."+
+						"Please set %s environment variable in your terminal or using --env flag", envVar.Name, envVar.Name)
+				}
+			}
+		}
+
+		app := apps.NewApp(config, spec)
+		if prepareEnvironment {
+			err := app.PrepareEnvironment()
+			if err != nil {
+				return err
+			}
+		}
+
+		specCommand, err := app.GetCommand(debug)
+		if err != nil {
+			return err
+		}
+
+		cmdio.LogString(ctx, "Running command: "+strings.Join(specCommand, " "))
+		appCmd := exec.Command(specCommand[0], specCommand[1:]...)
+		appCmd.Stdin = cmd.InOrStdin()
+		appCmd.Stdout = cmd.OutOrStdout()
+		appCmd.Stderr = cmd.ErrOrStderr()
+
+		appEnvs := apps.GetBaseEnvVars(config)
+		for _, envVar := range appEnvs {
+			env = append(env, envVar.String())
+		}
+		env = append(env, customEnv...)
+
+		appCmd.Env = env
+		appCmd.Dir = config.AppPath
+
+		err = appCmd.Start()
+		if err != nil {
+			return err
+		}
+
+		proxy, err := appproxy.New(ctx, config.AppURL)
+		if err != nil {
+			return err
+		}
+
+		me, err := w.CurrentUser.Me(ctx)
+		if err != nil {
+			return err
+		}
+
+		for key, value := range apps.GetXHeaders(me) {
+			proxy.InjectHeader(key, value)
+		}
+
+		proxyAddr := fmt.Sprintf("localhost:%d", port)
+		go func() {
+			cmdio.LogString(ctx, "To access your app go to http://"+proxyAddr)
+			err := proxy.ListenAndServe(proxyAddr)
+			if err != nil {
+				cmd.PrintErrln(err)
+			}
+		}()
+
+		if debug {
+			cmdio.LogString(ctx, "To debug your app, attach a debugger to port "+apps.DEBUG_PORT)
+		}
+
+		// Create a channel to handle graceful shutdown
+		done := make(chan error, 1)
+
+		// Handle interrupt signal
+		sigChan := make(chan os.Signal, 1)
+		signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)
```

Comment:
Please document both these signals (and that one is OS-independent and the other isn't).